[DOC] [PDF] Fix the way PDF build is set to use xelatex

jfbu opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->


#### Brief description of what is fixed or changed

The current way of using `xelatex` is wrong.  The `latex_engine` configuration value should be used. Else the Makefile put by Sphinx in builddir will be wrong and also the latexmkrc file setting for the Latexmk configuration.

The advice in doc/Makefile to use `LATEXMKOPTS='-xelatex -silent'` can be kept unchanged but is now only optional as the build will proceed with `xelatex` even without `-xelatex` (and anyhow it can not use anything else even prior to this PR). So the advice has value only about `-silent` now and one could (should?) redo the rules for `latexpdf` target to actually do the `make all-pdf` in builddir.

Setting `latex_engine` correctly avoids Sphinx putting in produced `.tex` file loading of packages such as `inputenc` which are irrelevant to `xelatex` and such extra packages may cause breakage in future depending on LaTeX installation, better to proceed only Sphinx docs instructions.


#### Other comments

The ``\defaultfontfeatures{Mapping=tex-text}`` line should probably be removed.  First it makes it incompatible to switching latex_engine to `'lualatex'`(syntax `Ligatures=TeX` should be used)) second the default Sphinx configuration has issued ``\defaultfontfeatures[\rmfamily,\sffamily,\ttfamily]{}`` for a reason, although perhaps here it is not important due to DejaVu font (untested).

For Sphinx related discussion see
[sphinx-doc/sphinx/pull/6888](https://github.com/sphinx-doc/sphinx/pull/6888)

The `latex_engine` configuration value was added at Sphinx Release 1.5 (released Dec 5, 2016).

#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

jfbu committed:
[DOC] [PDF] Fix the way PDF build is set to use xelatex

jfbu commented on doc/src/conf.py:
```diff
@@ -363,26 +363,20 @@
-\usepackage{amssymb}
-\usepackage{fontspec}
-\usepackage[english]{babel}
 \defaultfontfeatures{Mapping=tex-text}
```
This ``\defaultfontfeatures{Mapping=tex-text}`` should arguably be removed (cf https://github.com/sphinx-doc/sphinx/pull/6888) except if there is some good reason to keep it.

jfbu replied:
I have to research more at some point this https://github.com/sphinx-doc/sphinx/pull/6888, DejaVu has `tlig` feature, I have to check what fontspec does which transforms `"` into curly double quotes which is what https://github.com/sphinx-doc/sphinx/pull/6888 does not want, and make the `Mapping=tex-text` no good, but perhaps we can salvage other ligatures such as `ff`.

asmeurer replied:
Just to be clear, DejaVu Sans Mono only needs to be used for the monospace font (code examples). This is because it is the only font that renders Unicode pretty printed output correctly (for instance, compare this with DejaVu Sans Mono and other fonts)

```
⌠
⎮     1
⎮ ───────── dx
⎮   _______
⎮ ╲╱ x + 1
⌡
```

See https://github.com/sympy/sympy/issues/15700.

For body and heading text, the font can be anything. It would probably be best to use whatever Sphinx defaults to, or whatever makes the math look good.

jfbu replied:
Sphinx default to FreeSerif, FreeSans (GNU FreeFont).  The maths does look a bit small with DejaVu.  

I might try a FreeSerif build.  But in my fix #26883 to Missing characters I use that sans serif is DejaVu Sans.  This was some time-costly work and I am not eager to redo it from scratch to see what works with other font.

jfbu commented on doc/src/conf.py:
```diff
@@ -363,26 +363,20 @@
+    'babel': r'\usepackage[english]{babel}',
+    'fontpkg': r'''
 \usepackage{bm}
-\usepackage{amssymb}
```
This `\usepackage{bm}` seems to be there to allow `\bm` mark-up to obtain bold math.  Which is good, but I don't know if it is documented in Contributor guide and searching for it in the produced LaTeX file from `make latex`, I do not find a single example so I guess there are none in soures (it is easier to seach in monolithic LaTeX than across multiple sources via `grep -r`). Ping @asmeurer

asmeurer replied:
From the git blame, this comes from https://github.com/sympy/sympy/pull/1493

jfbu replied:
I traced the `\usepackage{bm}` seemingly first appearance in 
bed2d9332af410c67.  It was used before in b43b9dc256213 but for `pngmath_latex_preamble` which is not quite same thing (and the line `pngmath_divpng_args = ['-gamma 1.5','-D 110']` had a typo probably it should have been `pngmath_dvipng_args` as the binary is `dvipng` not `divpng` but it is a bit late in 2024 to review a 2009 commit...).  Anyhow in both cases the only reason for this is to allow `\bm` mark-up in math.  But from examining the (huge) LaTeX file, I see that there is currently not a single use of `\bm` in Sympy documentation.

Build succeeds after removal of `\usepackage{bm}`. I have no strong opinion on whether to keep it or not.  Many people use `\mathbf` in math with the hope it does what actually `\bm` is able of doing, not `\mathbf`.  As long as documentation does not mention `\bm` as recommended to contributors, there is no strong reason to keep it, but it is not a matter of primary importance.

jfbu replied:
A commit has since removed this `\usepackage{bm}`.

jfbu commented:
> As you can see from the `test / sphinx` CI, the pdf build fails with an error.

Indeed, the error is 
```
LaTeX Warning: Command \LaTeX invalid in math mode on input line 112.
```
which was probably triggered by the re-ordering it induces (I suspect something having to do with the fact that the displaced code worked by accident so far).  I could not build the PDF locally when I submitted the PR but now I can, so I will investigate and fix this.

asmeurer commented:
This stuff is all pretty old and no doubt predates a lot of improvements in Sphinx itself. I think really ideally this would all just work from the Sphinx end and we wouldn't have to do any LaTeX configuration at all (except for the DejaVu stuff, which is necessary to make SymPy Unicode pretty printed output render correctly).

jfbu commented on doc/src/conf.py:
```diff
@@ -363,26 +363,20 @@
-    'utf8extra': '',
     'preamble':  r'''
+% Define version of \LaTeX that is usable in math mode
+\let\OldLaTeX\LaTeX
```
```suggestion
\usepackage{letltxmacro}
\LetLtxMacro\OldLaTeX\LaTeX
```

jfbu replied:
Hmm, no. I will revert that. I did not want to spend so much time into debugging this alas I did.

First two side remarks:

1. The same issue I have at my locale with chromium are [seen](https://github.com/sympy/sympy/actions/runs/10167896565/job/28122085075?pr=26868#step:6:1574)
2. With current LaTeX there is no build crash with the OldLaTeX thing whatever its position in the preamble.

The test-sphinx here is done with TeXLive 2019, LaTeX of 2020-02-02.  It turns out that what matters is if the redefinition is done before or after `\usepackage[english]{babel}` is encountered.

The culprit is babel who did `\AtBeginDocument{\DeclareTextCommandDefault {\LaTeX }{\org@LaTeX }}` after having stored the meaning of `\LaTeX` into `\org@LaTeX`. When I moved the (doubtful because `\LaTeX` is a robust macro and was not handled as such) hack with  `\OldLaTeX` later in preamble, this caused `\org@LaTeX` to have preserved the old meaning which is the robust one: `\protect \LaTeX<space in the name>`. When the `\OldLaTeX` is earlier in preamble, the meaning of `\org@LaTeX` is `\text {\OldLaTeX }`. (which is what we want).

When I tried the `\LetLtxMacro` thinking there was some problem with robust macros and not wanting to investigate more, I thought issue was solved because it builds at my place and I never had made a complete build before (and I thought the problem came from toc file).  But the point is that with current babel and LaTeX and hyperref, things are different, the meaning is not modified by babel. Back with TeXLive 2019 as used for the GH action, the business with `\LetLtxMacro\OldLaTeX\LaTeX` served nothing anyhow, because `\LaTeX` was still redefined at begin document by babel to be `\protect\LaTeX<space in the name>` and `\LaTeX<space in the name>` was not modified by my code at all.  In current LaTeX, the action of `babel` has changed it does now `\DeclareTextCommandDefault {\LaTeX }{\ensureascii {\org@LaTeX }}` but it is nightmare to trace the LaTeX code at `\begin{document}` which occupies now tens of thousands of lines. Also `hyperref` has changed.  So it is very complicated to understand why with current LaTeX after `\begin{document}` we have `\LaTeX` expanding to `\text{\OldLaTeX}` as was expected.

Anyway in order to solve this on old LaTeX, we must make sure not only not to use `\let` (to avoid potential problems if `\LaTeX` is in a title ending up in `.aux` then `.toc` file) but also to delay at `\begin{document}` the redefinition of `\LaTeX` to override the fact that `babel` in those days redefined it there.

I will do that next.

jfbu reviewed (commented):
cf https://github.com/sympy/sympy/pull/26868#issuecomment-2258944163

jfbu committed:
Update doc/src/conf.py

Reason why original code worked in its earlier position in preamble remains to be investigated.

jfbu commented:
> This stuff is all pretty old and no doubt predates a lot of improvements in Sphinx itself. I think really ideally this would all just work from the Sphinx end and we wouldn't have to do any LaTeX configuration at all (except for the DejaVu stuff, which is necessary to make SymPy Unicode pretty printed output render correctly).

Basically this is exactly the intent of my PR using `latex_engine='xelatex'`.  But Sphinx uses `polygloassia` per default, so I left the `\usepackage[english]{babel}` thing, which could also be removed.

jfbu commented on doc/src/conf.py:
```diff
@@ -363,26 +363,21 @@
+% Define version of \LaTeX that is usable in math mode
+\usepackage{letltxmacro}
+\LetLtxMacro\OldLaTeX\LaTeX
+\renewcommand{\LaTeX}{\text{\OldLaTeX}}
```
```suggestion
\AtBeginDocument{\DeclareRobustCommand{\LaTeX}{\text{\OldLaTeX}}}
```

jfbu committed:
Update doc/src/conf.py to get \LaTeX redefinition overcome babel of the past

jfbu committed:
Update LaTeX requirements for PDF build via xelatex

jfbu commented on doc/aptinstall.sh:
```diff
@@ -18,6 +18,8 @@ sudo apt-get install\
   texlive-fonts-extra\
   texlive-xetex\
+  xindy\
+  xindy-rules\
```
I hesitated whether I should update `doc/src/contributing/new-contributors-guide/build-docs.rst` as well, because it does not seem complete already (no mention of `texlive-xetex` there).

An alternative is to use `latex_use_xindy = False` in conf.py. This will keep using `makeindex` so the Index will look same as so far.  Seemingly there aren't Unicode characters in there.  Hmm maybe it is better I push this rather.  Only waiting to see if test-sphinx succeeds with xindy, then I will use `latex_use_xindy = False`.

jfbu replied:
The [test-sphinx](https://github.com/sympy/sympy/actions/runs/10170020818/job/28128916368?pr=26868) succeeded with xindy.

For reasons of not adding dependencies needing documentation I will push `latex_use_xindy = False` rather.

asmeurer replied:
You can also update the build docs page. The pdf build instructions aren't maintained as well because not very many people build the pdf docs, but there's no reason they shouldn't be accurate.

jfbu replied:
After pondering the matter I simpliy pushed `latex_use_xindy = False` thus sparing me the additional updates to build docs and dependencies. I would never have been able anyhow to document what is needed on Windows.

asmeurer replied:
I didn't even realize the pdf docs could be built on Windows. It isn't a high priority if we need some package to make them better. Most people don't need to build the pdf docs locally. It's mainly just important that they can be built on the CI so that they get compiled when we do a release (plus the PDF docs have a the nice side effect of catching invalid LaTeX math examples).

jfbu committed:
Revert "Update LaTeX requirements for PDF build via xelatex"

This reverts commit cef36fd4a1606da1435a7c8e6798157a29e76018.

jfbu committed:
Keep using makeindex, not xindy, for xelatex builds

jfbu committed:
Trim latex_elements of unneeded remnants

jfbu committed:
Update .mailmap for Jean-François B.

jfbu committed:
[DOC] Fix doc/Makefile, setting PAPER had no effect

jfbu committed:
[DOC] Better Makefile for PDF via xelatex production

jfbu committed:
[DOC] latex target must require logo as dependency

jfbu committed:
[DOC] Add @ prefix to sub-make in _build/latex from pdf target

jfbu commented on doc/Makefile:
```diff
@@ -7,7 +7,8 @@ RST2HTML     = rst2html
 SPHINXBUILD  = sphinx-build
-PAPER        =
+# This one can also be set in the environment
+PAPER        ?= letter
```
There was a typo so this PAPER Makefile variable had no effect.  Sphinx defaults to using 'letterpaper', hence the update. I think  it makes sense to allow also setting PAPER in environment not only command line hence the `?`.

jfbu commented on doc/Makefile:
```diff
@@ -16,15 +17,11 @@ LIVEPORT     = 0
 SPHINXSAVEWARNINGS = -w $(WARNINGSFILE)
 ALLSPHINXOPTSapi = -d $(BUILDDIR)/doctrees-api $(SPHINXOPTS) api
-ALLSPHINXOPTSlatex = -d $(BUILDDIR)/doctrees-latex -D latex_element.papersize=$(PAPER)paper \
+ALLSPHINXOPTSlatex = -d $(BUILDDIR)/doctrees-latex -D latex_elements.papersize=$(PAPER)paper \
```
This is the fix for the PAPER typo.

jfbu commented on doc/Makefile:
```diff
@@ -104,16 +101,18 @@ htmlhelp:
 	      ".hhp project file in $(BUILDDIR)/htmlhelp."
 
-latex:
+latex: logo
```
On a fresh context with no prior `make html`,  the `make latex` without logo requirement produces a `.tex` file which causes `xelatex` to crash.

asmeurer replied:
As long as we're fixing this, it looks like it's configured incorrectly. This will make it run the first time, but if `make logo` fails it doesn't recognize that and won't run it again. For me it failed with

```
ERROR:root:Return code is not 0: Command: convert /Users/aaronmeurer/Documents/Python/sympy/sympy/doc/./_build/logo/sympy-16px.png /Users/aaronmeurer/Documents/Python/sympy/sympy/doc/./_build/logo/sympy-32px.png /Users/aaronmeurer/Documents/Python/sympy/sympy/doc/./_build/logo/sympy-48px.png /Users/aaronmeurer/Documents/Python/sympy/sympy/doc/./_build/logo/sympy-64px.png /Users/aaronmeurer/Documents/Python/sympy/sympy/doc/./_build/logo/sympy-favicon.ico
```

(my `convert` command is broken on my machine)

jfbu replied:
ah I did not notice because fortunately for me`make logo` works fine.  My `convert` is from the Fink package manager for macOS and is a bit old ``Version: ImageMagick 7.0.11-13 Q16 x86_64 2021-05-17 https://imagemagick.org``, but I update fink packages once a year at most. I am lucky my old system is still supported but I may reach end of of my luck next time.

Should I keep the `logo` here in view of your remark? I am not sure what this implies for this. Most people will build HTML in priority, and once done, the images are in place for `make latex` to pick them, (if `make logo` succeeded).

asmeurer replied:
Yes, it should be here. I was just pointing out that the `logo` target itself is not properly depending on its outputs.

jfbu commented on doc/Makefile:
```diff
@@ -104,16 +101,18 @@ htmlhelp:
 
 pdf pdflatex latexpdf: latex
-	$(MAKE) -C $(BUILDDIR)/latex
+	@$(MAKE) -C $(BUILDDIR)/latex all-pdf \
+	LATEXOPTS="-halt-on-error -interaction=batchmode" \
+	LATEXMKOPTS="-xelatex" \
+	|| echo "LaTeX error"'!'" Check .log in $(BUILDDIR)/latex for lineno in .tex file."
```
This is the correct way, now that the Sphinx Makefile and latexmkrc are correctly configured, to use LATEXOPTS and LATEXMKOPTS. The `-silent` is too drastic for Latexmk in case of problem, or for example when Latexmk does many successive passes, the reasons get lost. So we pass batchmode to the xelatex excutable rather to keep avoiding humongous console output.

jfbu committed:
[DOC] Fix a glitch in latex Makefile target

jfbu commented on doc/Makefile:
```diff
@@ -104,16 +101,18 @@ htmlhelp:
-latex:
-	mkdir -p $(BUILDDIR)/latex $(BUILDDIR)/doctrees
+latex: logo
+	mkdir -p $(BUILDDIR)/latex $(BUILDDIR)/doctrees-latex
```
The `-latex` was missing but of course this did not cause harm.

Note that the entire line could/should be removed for this and other similar build targets, because `sphinx-build` does not need it.  Maybe it did in the distant path.

jfbu commented:
I fixed a few pre-existing things in the Makefile.  I dropped the `-silent` option of Latexmk because it is too radical, in favor of only using "batchmode" for the `xelatex` builds and not suppress the Latexmk own diagnostics.

One can see the effect of the `-xelatex` Latexmk option: during the three passes, `xelatex` produces an `.xdv` not a pdf file.  The pdf file is produced at very end by `xdvipdfmx` (which is also used internally by `xelatex` if used to produce pdf directly).

Although we at Sphinx have been indicating for some years the `-xelatex` as improving speed, [recent testing](https://github.com/sphinx-doc/sphinx/pull/12636) shows this is not really true. It would be true for documents doing great numbers of graphics inclusion. The most important in terms of speed is to keep using TeXLive 2019 based LaTeX, the 2024 XeLaTeX is at least 50% slower.

The advantage of `.xdv` first then `.pdf` is also that in case of an error which stops LaTeX (see `-halt-on-error`) the  partial `.xdv` file is still there and often `xdvipdfmx` can be used on it, which gives an indication of what went smoothly prior to error.  And TeX always indicates in the log file the input line number where the macro was encountered causing an error.  The Makefile rule is structured to tell that to user, for example, here is a run with  an error:

```text
.... lines suppressed ...
------------
Run number 1 of rule 'xelatex'
------------
------------
Running 'xelatex --no-pdf -halt-on-error -interaction=batchmode -no-pdf -recorder  "sympy-1.14.dev.tex"'
------------
This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
 restricted \write18 enabled.
entering extended mode
Latexmk: Getting log file 'sympy-1.14.dev.log'
Latexmk: Examining 'sympy-1.14.dev.fls'
Latexmk: Examining 'sympy-1.14.dev.log'
Latexmk: Index file 'sympy-1.14.dev.idx' was written
Latexmk: Log file says output to 'sympy-1.14.dev.xdv'
Have index file 'sympy-1.14.dev.idx', sympy-1.14.dev.ind sympy-1.14.dev
Latexmk: Errors, so I did not complete making targets
Collected error summary (may duplicate other messages):
  xelatex: Command for 'xelatex' gave return code 1
      Refer to 'sympy-1.14.dev.log' and/or above output for details

Latexmk: Sometimes, the -f option can be used to get latexmk
  to try to force complete processing.
  But normally, you will need to correct the file(s) that caused the
  error, and then rerun latexmk.
  In some cases, it is best to clean out generated files before rerunning
  latexmk after you've corrected the files.
make[1]: *** [Makefile:28: sympy-1.14.dev.pdf] Error 12
make[1] : on quitte le répertoire « /pathtowork/_gits/sympy/doc/_build/latex »
LaTeX error! Check .log in _build/latex for lineno in .tex file.
```

In the case at hand the error came from a previous babel run left-over aux file, I had not cleaned the build dir, and now the Sphinx default of using polyglossia is applied.

Here is the complete console output from a successful `make latexpdf`,  dropping the `make latex` part.  This is localized to French.  The xdvipdfmx warning about links and page boundaries are a priori benign.  One does not see them when one does not use `-xelatex` option and each `xelatex` pass produces a PDF directly, but as far as I know this is only  a different user interaction, because I have always been told that when `xelatex` produces a PDF directly it first does the `.xdv` then `xdvipdfmx` and removes the `.xdv`.

```text
Les fichiers LaTeX se trouvent dans _build/latex.
Exécuter 'make' dans ce répertoire pour les soumettre à (pdf)latex
(ou 'make latexpdf' directement ici pour l’automatiser).

Build finished; the LaTeX files are in _build/latex.
Run `make all' in that directory to run these through xelatex.
make[1] : on entre dans le répertoire « /pathtowork/sympy/doc/_build/latex »
latexmk -pdf -dvi- -ps- -xelatex 'sympy-1.14.dev.tex'
Rc files read:
  latexmkrc
Latexmk: This is Latexmk, John Collins, 7 Apr. 2024. Version 4.85.
No existing .aux file, so I'll make a simple one, and require run of *latex.
Latexmk: applying rule 'xelatex'...
Rule 'xelatex':  Reasons for rerun
Category 'other':
  Rerun of 'xelatex' forced or previously required:
    Reason or flag: 'Initial setup'

------------
Run number 1 of rule 'xelatex'
------------
------------
Running 'xelatex --no-pdf -halt-on-error -interaction=batchmode -no-pdf -recorder  "sympy-1.14.dev.tex"'
------------
This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
 restricted \write18 enabled.
entering extended mode
Latexmk: Getting log file 'sympy-1.14.dev.log'
Latexmk: Examining 'sympy-1.14.dev.fls'
Latexmk: Examining 'sympy-1.14.dev.log'
Latexmk: Index file 'sympy-1.14.dev.idx' was written
Latexmk: References changed.
Latexmk: Missing input file 'sympy-1.14.dev.toc' (or dependence on it) from following:
  No file sympy-1.14.dev.toc.
Latexmk: Missing input file 'sympy-1.14.dev.ind' (or dependence on it) from following:
  No file sympy-1.14.dev.ind.
Latexmk: References changed.
Latexmk: References changed.
Latexmk: Log file says output to 'sympy-1.14.dev.xdv'
Have index file 'sympy-1.14.dev.idx', sympy-1.14.dev.ind sympy-1.14.dev
Latexmk: applying rule 'makeindex sympy-1.14.dev.idx'...
Rule 'makeindex sympy-1.14.dev.idx':  Reasons for rerun
Category 'other':
  Rerun of 'makeindex sympy-1.14.dev.idx' forced or previously required:
    Reason or flag: 'Initial set up of rule'

------------
Run number 1 of rule 'makeindex sympy-1.14.dev.idx'
------------
------------
Running 'makeindex -s python.ist  -o "sympy-1.14.dev.ind" "sympy-1.14.dev.idx"'
------------
This is makeindex, version 2.17 [TeX Live 2024] (kpathsea + Thai support).
Scanning style file ./python.ist.......done (7 attributes redefined, 0 ignored).
Scanning input file sympy-1.14.dev.idx..........done (6624 entries accepted, 0 rejected).
Sorting entries.................................................................done (92666 comparisons).
Generating output file sympy-1.14.dev.ind..........done (9777 lines written, 0 warnings).
Output written in sympy-1.14.dev.ind.
Transcript written in sympy-1.14.dev.ilg.
Latexmk: applying rule 'xelatex'...
Rule 'xelatex':  Reasons for rerun
Changed files or newly in use/created:
  sympy-1.14.dev.aux
  sympy-1.14.dev.ind
  sympy-1.14.dev.out
  sympy-1.14.dev.toc

------------
Run number 2 of rule 'xelatex'
------------
------------
Running 'xelatex --no-pdf -halt-on-error -interaction=batchmode -no-pdf -recorder  "sympy-1.14.dev.tex"'
------------
This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
 restricted \write18 enabled.
entering extended mode
Latexmk: Getting log file 'sympy-1.14.dev.log'
Latexmk: Examining 'sympy-1.14.dev.fls'
Latexmk: Examining 'sympy-1.14.dev.log'
Latexmk: Index file 'sympy-1.14.dev.idx' was written
Latexmk: References changed.
Latexmk: References changed.
Latexmk: Log file says output to 'sympy-1.14.dev.xdv'
Have index file 'sympy-1.14.dev.idx', sympy-1.14.dev.ind sympy-1.14.dev
Latexmk: applying rule 'makeindex sympy-1.14.dev.idx'...
Rule 'makeindex sympy-1.14.dev.idx':  Reasons for rerun
Changed files or newly in use/created:
  sympy-1.14.dev.idx

------------
Run number 2 of rule 'makeindex sympy-1.14.dev.idx'
------------
------------
Running 'makeindex -s python.ist  -o "sympy-1.14.dev.ind" "sympy-1.14.dev.idx"'
------------
This is makeindex, version 2.17 [TeX Live 2024] (kpathsea + Thai support).
Scanning style file ./python.ist.......done (7 attributes redefined, 0 ignored).
Scanning input file sympy-1.14.dev.idx..........done (6624 entries accepted, 0 rejected).
Sorting entries.................................................................done (92666 comparisons).
Generating output file sympy-1.14.dev.ind..........done (9777 lines written, 0 warnings).
Output written in sympy-1.14.dev.ind.
Transcript written in sympy-1.14.dev.ilg.
Latexmk: applying rule 'xelatex'...
Rule 'xelatex':  Reasons for rerun
Changed files or newly in use/created:
  sympy-1.14.dev.aux
  sympy-1.14.dev.ind
  sympy-1.14.dev.out
  sympy-1.14.dev.toc

------------
Run number 3 of rule 'xelatex'
------------
------------
Running 'xelatex --no-pdf -halt-on-error -interaction=batchmode -no-pdf -recorder  "sympy-1.14.dev.tex"'
------------
This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
 restricted \write18 enabled.
entering extended mode
Latexmk: Getting log file 'sympy-1.14.dev.log'
Latexmk: Examining 'sympy-1.14.dev.fls'
Latexmk: Examining 'sympy-1.14.dev.log'
Latexmk: Index file 'sympy-1.14.dev.idx' was written
Latexmk: Log file says output to 'sympy-1.14.dev.xdv'
Have index file 'sympy-1.14.dev.idx', sympy-1.14.dev.ind sympy-1.14.dev
Latexmk: applying rule 'xdvipdfmx'...
Rule 'xdvipdfmx':  Reasons for rerun
Changed files or newly in use/created:
  sympy-1.14.dev.xdv
Category 'changed_source_rules':
  xelatex
Category 'no_dest':
  xdvipdfmx

------------
Run number 1 of rule 'xdvipdfmx'
------------
------------
Running 'xdvipdfmx -E -o "sympy-1.14.dev.pdf"  "sympy-1.14.dev.xdv"'
------------
sympy-1.14.dev.xdv -> sympy-1.14.dev.pdf
[1][2][3][4][5][6][7][8][9][10][11][12][13][14][15][16][17][18][19][20][21][22][23][24][25][26][27][28][29][30][31][32][33][34][35][36][37][38][39][40][41][42][43][44][45][46][47][48][49][50][51][52][53][54][55][56][57][58][59][60][61][62][63][64][65
xdvipdfmx:warning: Trying to include PDF file with version (1.7), which is newer than current output PDF setting (1.5).
][66][67
xdvipdfmx:warning: Trying to include PDF file with version (1.7), which is newer than current output PDF setting (1.5).
xdvipdfmx:warning: Trying to include PDF file with version (1.7), which is newer than current output PDF setting (1.5).
][68
xdvipdfmx:warning: Trying to include PDF file with version (1.7), which is newer than current output PDF setting (1.5).
][69][70][71][72][73][74][75
xdvipdfmx:warning: PDF file is tagged... Ignoring tags.
][76][77][78][79][80][81][82][83][84][85][86][87][88][89][90][91][92][93][94][95][96][97][98][99][100][101][102][103][104][105][106][107][108][109][110][111][112][113][114][115][116][117][118][119][120][121][122][123][124][125][126][127][128][129][130][131][132][133][134][135][136][137][138][139][140][141][142][143][144][145][146][147][148][149][150][151][152][153][154][155][156][157][158][159][160][161][162][163][164][165][166][167][168][169][170][171][172][173
xdvipdfmx:warning: PDF file is tagged... Ignoring tags.
][174][175
xdvipdfmx:warning: PDF file is tagged... Ignoring tags.
][176][177][178][179][180][181
xdvipdfmx:warning: PDF file is tagged... Ignoring tags.
][182][183
xdvipdfmx:warning: PDF file is tagged... Ignoring tags.
][184][185][186][187][188][189][190][191][192][193][194][195][196][197][198][199][200][201][202][203][204][205][206][207][208][209][210][211][212][213][214][215][216][217][218][219][220][221][222][223][224][225][226][227][228][229][230][231][232][233][234][235][236][237][238][239][240][241][242][243][244][245][246][247][248][249][250][251][252][253][254][255][256][257][258][259][260][261][262][263][264][265][266][267][268][269][270][271][272][273][274][275][276][277][278][279][280][281][282][283][284][285][286][287][288][289][290][291][292][293][294][295][296][297][298][299][300][301][302][303][304][305][306][307][308][309][310][311][312][313][314][315][316][317][318][319][320][321][322][323][324][325][326][327][328][329][330][331][332][333][334][335][336][337][338][339][340][341][342][343][344][345
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [403.146 -432.872 474.577 -421.275]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [403.146 -564.379 447.73 -552.782]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [403.146 -648.065 436.279 -636.468]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [403.146 -767.617 432.616 -756.02]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [403.146 -803.482 440.555 -791.885]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [514.832 -839.348 534.022 -827.751]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [389.978 -851.303 463.724 -839.706]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [402.747 -923.034 441.61 -911.437]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [403.146 -946.944 450.24 -935.347]
xdvipdfmx:warning: Maybe incorrect paper size specified.
][346][347][348][349][350][351][352][353][354][355][356][357][358][359][360][361][362][363][364][365][366][367][368][369][370][371][372][373][374][375][376][377][378][379][380][381][382][383][384][385][386][387][388][389][390][391][392][393][394][395][396][397][398][399][400][401][402][403][404][405][406][407][408][409][410][411][412][413][414][415][416][417][418][419][420][421][422][423][424][425][426][427][428][429][430][431][432][433][434][435][436][437][438][439][440][441][442][443][444][445][446][447][448][449][450][451][452][453][454][455][456][457][458][459][460][461][462][463][464][465][466][467][468][469][470][471][472][473][474][475][476][477][478][479][480][481][482][483][484][485][486][487][488][489][490][491][492][493][494][495][496][497][498][499][500][501][502][503][504][505][506][507][508][509][510][511][512][513][514][515][516][517][518][519][520][521][522][523][524][525][526][527][528][529][530][531][532][533][534][535][536][537][538][539][540][541][542][543][544][545][546][547][548][549][550][551][552][553][554][555][556][557][558][559][560][561][562][563][564][565][566][567][568][569][570][571][572][573][574][575][576][577][578][579][580][581][582][583][584][585][586][587][588][589][590][591][592][593][594][595][596][597][598][599][600][601][602][603][604][605][606][607][608][609][610][611][612][613][614][615][616][617][618][619][620][621][622][623][624][625][626][627][628][629][630][631][632][633][634][635][636][637][638][639][640][641][642][643][644][645][646][647][648][649][650][651][652][653][654][655][656][657][658][659][660][661][662][663][664][665][666][667][668][669][670][671][672][673][674][675][676][677][678][679][680][681][682][683][684][685][686][687][688][689][690][691][692][693][694][695][696][697][698][699][700][701][702][703][704][705][706][707][708][709][710][711][712][713][714][715][716][717][718][719][720][721][722][723][724][725][726][727][728][729][730][731][732][733][734][735][736][737][738][739][740][741][742][743][744][745][746][747][748][749][750][751][752][753][754][755][756][757][758][759][760][761][762][763][764][765][766][767][768][769][770][771][772][773][774][775][776][777][778][779][780][781][782][783][784][785][786][787][788][789][790][791][792][793][794][795][796][797][798][799][800][801][802][803][804][805][806][807][808][809][810][811][812][813][814][815][816][817][818][819][820][821][822][823][824][825][826][827][828][829][830][831][832][833][834][835][836][837][838][839][840][841][842][843][844][845][846][847][848][849][850][851][852][853][854][855][856][857][858][859][860][861][862][863][864][865][866][867][868][869][870][871][872][873][874][875][876][877][878][879][880][881][882][883][884][885][886][887][888][889][890][891][892][893][894][895][896][897][898][899][900][901][902][903][904][905][906][907][908][909][910][911][912][913][914][915][916][917][918][919][920][921][922][923][924][925][926][927][928][929][930][931][932][933][934][935][936][937][938][939][940][941][942][943][944][945][946][947][948][949][950][951][952][953][954][955][956][957][958][959][960][961][962][963][964][965][966][967][968][969][970][971][972][973][974][975][976][977][978][979][980][981][982][983][984][985][986][987][988][989][990][991][992][993][994][995][996][997][998][999][1000][1001][1002][1003][1004][1005][1006][1007][1008][1009][1010][1011][1012][1013][1014][1015][1016][1017][1018][1019][1020][1021][1022][1023][1024][1025][1026][1027][1028][1029][1030][1031][1032][1033][1034][1035][1036][1037][1038][1039][1040][1041][1042][1043][1044][1045][1046][1047][1048][1049][1050][1051][1052][1053][1054][1055][1056][1057][1058][1059][1060][1061][1062][1063][1064][1065][1066][1067][1068][1069][1070][1071][1072][1073][1074][1075][1076][1077][1078][1079][1080][1081][1082][1083][1084][1085][1086][1087][1088][1089][1090][1091][1092][1093][1094][1095][1096][1097][1098][1099][1100][1101][1102][1103][1104][1105][1106][1107][1108][1109][1110][1111][1112][1113][1114][1115][1116][1117][1118][1119][1120][1121][1122][1123][1124][1125][1126][1127][1128][1129][1130][1131][1132][1133][1134][1135][1136][1137][1138][1139][1140][1141][1142][1143][1144][1145][1146][1147][1148][1149][1150][1151][1152][1153][1154][1155][1156][1157][1158][1159][1160][1161][1162][1163][1164][1165][1166][1167][1168][1169][1170][1171][1172][1173][1174][1175][1176][1177][1178][1179][1180][1181][1182][1183][1184][1185][1186][1187][1188][1189][1190][1191][1192][1193][1194][1195][1196][1197][1198][1199][1200][1201][1202][1203][1204][1205][1206][1207][1208][1209][1210][1211][1212][1213][1214][1215][1216][1217][1218][1219][1220][1221][1222][1223][1224][1225][1226][1227][1228][1229][1230][1231][1232][1233][1234][1235][1236][1237][1238][1239][1240][1241][1242][1243][1244][1245][1246][1247][1248][1249][1250][1251][1252][1253][1254][1255][1256][1257][1258][1259][1260][1261][1262][1263][1264][1265][1266][1267][1268][1269][1270][1271][1272][1273][1274][1275][1276][1277][1278][1279][1280][1281][1282][1283][1284][1285][1286][1287][1288][1289][1290][1291][1292][1293][1294][1295][1296][1297][1298][1299][1300][1301][1302][1303][1304][1305][1306][1307][1308][1309][1310][1311][1312][1313][1314][1315][1316][1317][1318][1319][1320][1321][1322][1323][1324][1325][1326][1327][1328][1329][1330][1331][1332][1333][1334][1335][1336][1337][1338][1339][1340][1341][1342][1343][1344][1345][1346][1347][1348][1349][1350][1351][1352][1353][1354][1355][1356][1357][1358][1359][1360][1361][1362][1363][1364][1365][1366][1367][1368][1369][1370][1371][1372][1373][1374][1375][1376][1377][1378][1379][1380][1381][1382][1383][1384][1385][1386][1387][1388][1389][1390][1391][1392][1393][1394][1395][1396][1397][1398][1399][1400][1401][1402][1403][1404][1405][1406][1407][1408][1409][1410][1411][1412][1413][1414][1415][1416][1417][1418][1419][1420][1421][1422][1423][1424][1425][1426][1427][1428][1429][1430][1431][1432][1433][1434][1435][1436][1437][1438][1439][1440][1441][1442][1443][1444][1445][1446][1447][1448][1449][1450][1451][1452][1453][1454][1455][1456][1457][1458][1459][1460][1461][1462][1463][1464][1465][1466][1467][1468][1469][1470][1471][1472][1473][1474][1475][1476][1477][1478][1479][1480][1481][1482][1483][1484][1485][1486][1487][1488][1489][1490][1491][1492][1493][1494][1495][1496][1497][1498][1499][1500][1501][1502][1503][1504][1505][1506][1507][1508][1509][1510][1511][1512][1513][1514][1515][1516][1517][1518][1519][1520][1521][1522][1523][1524][1525][1526][1527][1528][1529][1530][1531][1532][1533][1534][1535][1536][1537][1538][1539][1540][1541][1542][1543][1544][1545][1546][1547][1548][1549][1550][1551][1552][1553][1554][1555][1556][1557][1558][1559][1560][1561][1562][1563][1564][1565][1566][1567][1568][1569][1570][1571][1572][1573][1574][1575][1576][1577][1578][1579][1580][1581][1582][1583][1584][1585][1586][1587][1588][1589][1590][1591][1592][1593][1594][1595][1596][1597][1598][1599][1600][1601][1602][1603][1604][1605][1606][1607][1608][1609][1610][1611][1612][1613][1614][1615][1616][1617][1618][1619][1620][1621][1622][1623][1624][1625][1626][1627][1628][1629][1630][1631][1632][1633][1634][1635][1636][1637][1638][1639][1640][1641][1642][1643][1644][1645][1646][1647][1648][1649][1650][1651][1652][1653][1654][1655][1656][1657][1658][1659][1660][1661][1662][1663][1664][1665][1666][1667][1668][1669][1670][1671][1672][1673][1674][1675][1676][1677][1678][1679][1680][1681][1682][1683][1684][1685][1686][1687][1688][1689][1690][1691][1692][1693][1694][1695][1696][1697][1698][1699][1700][1701][1702][1703][1704][1705][1706][1707][1708][1709][1710][1711][1712][1713][1714][1715][1716][1717][1718][1719][1720][1721][1722][1723][1724][1725][1726][1727][1728][1729][1730][1731][1732][1733][1734][1735][1736][1737][1738][1739][1740][1741][1742][1743][1744][1745][1746][1747][1748][1749][1750][1751][1752][1753][1754][1755][1756][1757][1758][1759][1760][1761][1762][1763][1764][1765][1766][1767][1768][1769][1770][1771][1772][1773][1774][1775][1776][1777][1778][1779][1780][1781][1782][1783][1784][1785][1786][1787][1788][1789][1790][1791][1792][1793][1794][1795][1796][1797][1798][1799][1800][1801][1802][1803][1804][1805][1806][1807][1808][1809][1810][1811][1812][1813][1814][1815][1816][1817][1818][1819][1820][1821][1822][1823][1824][1825][1826][1827][1828][1829][1830][1831][1832][1833][1834][1835][1836][1837][1838][1839][1840][1841][1842][1843][1844][1845][1846][1847][1848][1849][1850][1851][1852][1853][1854][1855][1856][1857][1858][1859][1860][1861][1862][1863][1864][1865][1866][1867][1868][1869][1870][1871][1872
xdvipdfmx:warning: PDF file is tagged... Ignoring tags.
][1873][1874][1875][1876][1877
xdvipdfmx:warning: PDF file is tagged... Ignoring tags.
][1878][1879][1880][1881][1882][1883][1884][1885][1886][1887
xdvipdfmx:warning: PDF file is tagged... Ignoring tags.
][1888][1889][1890][1891][1892][1893][1894][1895][1896][1897][1898][1899][1900][1901][1902][1903][1904][1905][1906][1907][1908][1909][1910][1911][1912][1913][1914][1915][1916][1917][1918][1919][1920][1921][1922][1923][1924][1925][1926][1927][1928][1929][1930][1931][1932][1933][1934][1935][1936][1937][1938][1939][1940][1941][1942][1943][1944][1945][1946][1947][1948][1949][1950][1951][1952][1953][1954][1955][1956][1957][1958][1959][1960][1961][1962][1963][1964][1965][1966][1967][1968][1969][1970][1971][1972][1973][1974][1975][1976][1977][1978][1979][1980][1981][1982][1983][1984][1985][1986][1987][1988][1989][1990][1991][1992][1993][1994][1995][1996][1997][1998][1999][2000][2001][2002][2003][2004][2005][2006][2007][2008][2009][2010][2011][2012][2013][2014][2015][2016][2017][2018][2019][2020][2021][2022][2023][2024][2025][2026][2027][2028][2029][2030][2031][2032][2033][2034][2035][2036][2037][2038][2039][2040][2041][2042][2043][2044][2045][2046][2047][2048][2049][2050][2051][2052][2053][2054][2055][2056][2057][2058][2059][2060][2061][2062][2063][2064][2065][2066][2067][2068][2069][2070][2071][2072][2073][2074][2075][2076][2077][2078][2079][2080][2081][2082][2083][2084][2085][2086][2087][2088][2089][2090][2091][2092][2093][2094][2095][2096][2097][2098][2099][2100][2101][2102][2103][2104][2105][2106][2107][2108][2109][2110][2111][2112][2113][2114][2115][2116][2117][2118][2119][2120][2121][2122][2123][2124][2125][2126][2127][2128][2129][2130][2131][2132][2133][2134][2135][2136][2137][2138][2139][2140][2141][2142][2143][2144][2145][2146][2147][2148][2149][2150][2151][2152][2153][2154][2155][2156][2157][2158][2159][2160][2161][2162][2163][2164][2165][2166][2167][2168][2169][2170][2171][2172][2173][2174][2175][2176][2177][2178][2179][2180][2181][2182][2183][2184][2185][2186][2187][2188][2189][2190][2191][2192][2193][2194][2195][2196][2197][2198][2199][2200][2201][2202][2203][2204][2205][2206][2207][2208][2209][2210][2211][2212][2213][2214][2215][2216][2217][2218][2219][2220][2221][2222][2223][2224][2225][2226][2227][2228][2229][2230][2231][2232][2233][2234][2235][2236][2237][2238][2239][2240][2241][2242][2243][2244][2245][2246][2247][2248][2249][2250][2251][2252][2253][2254][2255][2256][2257][2258][2259][2260][2261][2262][2263][2264][2265][2266][2267][2268][2269][2270][2271][2272][2273][2274][2275][2276][2277][2278][2279][2280][2281][2282][2283][2284][2285][2286][2287][2288][2289][2290][2291][2292][2293][2294][2295][2296][2297][2298][2299][2300][2301][2302][2303][2304][2305][2306][2307][2308][2309][2310][2311][2312][2313][2314][2315][2316][2317][2318][2319][2320][2321][2322][2323][2324][2325][2326][2327][2328][2329][2330][2331][2332][2333][2334][2335][2336][2337][2338][2339][2340][2341][2342][2343][2344][2345][2346][2347][2348][2349][2350][2351][2352][2353][2354][2355][2356
xdvipdfmx:warning: Trying to include PDF file with version (1.7), which is newer than current output PDF setting (1.5).
][2357][2358][2359][2360][2361][2362][2363][2364][2365][2366][2367][2368][2369][2370][2371][2372][2373][2374][2375][2376][2377][2378][2379][2380
xdvipdfmx:warning: Trying to include PDF file with version (1.7), which is newer than current output PDF setting (1.5).
][2381][2382][2383][2384][2385][2386][2387][2388][2389][2390][2391][2392][2393][2394][2395][2396][2397][2398][2399][2400][2401][2402][2403][2404][2405][2406][2407][2408][2409][2410][2411][2412][2413][2414][2415][2416][2417][2418][2419][2420][2421][2422][2423][2424][2425][2426][2427][2428][2429][2430][2431][2432][2433][2434][2435][2436][2437][2438][2439][2440][2441][2442][2443][2444][2445][2446][2447][2448][2449][2450][2451][2452][2453][2454][2455][2456][2457][2458][2459][2460][2461][2462][2463][2464][2465][2466][2467][2468][2469][2470][2471][2472][2473][2474][2475][2476][2477][2478][2479][2480][2481][2482][2483][2484][2485][2486][2487][2488][2489][2490][2491][2492][2493][2494][2495][2496][2497][2498][2499][2500][2501][2502][2503][2504][2505][2506][2507][2508][2509][2510][2511][2512][2513][2514][2515][2516][2517][2518][2519][2520][2521][2522][2523][2524][2525][2526][2527][2528][2529][2530][2531][2532][2533][2534][2535][2536][2537][2538][2539][2540][2541][2542][2543][2544][2545][2546][2547][2548][2549][2550][2551][2552][2553][2554][2555][2556][2557][2558][2559][2560][2561][2562][2563][2564][2565][2566][2567][2568][2569][2570][2571][2572][2573][2574][2575][2576][2577][2578][2579][2580][2581][2582][2583][2584][2585][2586][2587][2588][2589][2590][2591][2592][2593][2594][2595][2596][2597][2598][2599][2600][2601][2602][2603][2604][2605][2606][2607][2608][2609][2610][2611][2612][2613][2614][2615][2616][2617][2618][2619][2620][2621][2622][2623][2624][2625][2626][2627][2628][2629][2630][2631][2632][2633][2634][2635][2636][2637][2638][2639][2640][2641][2642][2643][2644][2645][2646][2647][2648][2649][2650][2651][2652][2653][2654][2655][2656][2657][2658][2659][2660][2661][2662][2663][2664][2665][2666][2667][2668][2669][2670][2671][2672][2673][2674][2675][2676][2677][2678][2679][2680][2681][2682][2683][2684][2685][2686][2687][2688][2689][2690][2691][2692][2693][2694][2695][2696][2697][2698][2699][2700][2701][2702][2703][2704][2705][2706][2707][2708][2709][2710][2711][2712][2713][2714][2715][2716][2717][2718][2719][2720][2721][2722][2723][2724][2725][2726][2727][2728][2729][2730][2731][2732][2733][2734][2735][2736][2737][2738][2739][2740][2741][2742][2743][2744][2745][2746][2747][2748][2749][2750][2751][2752][2753][2754][2755][2756][2757][2758][2759][2760][2761][2762][2763][2764][2765][2766][2767][2768][2769][2770][2771][2772][2773][2774][2775][2776][2777][2778][2779][2780][2781][2782][2783][2784][2785][2786][2787][2788][2789][2790][2791][2792][2793][2794][2795][2796][2797][2798][2799][2800][2801][2802][2803][2804][2805][2806][2807][2808][2809][2810][2811][2812][2813][2814][2815][2816][2817][2818][2819][2820][2821][2822][2823][2824][2825][2826][2827][2828][2829][2830][2831][2832][2833][2834][2835][2836][2837][2838][2839][2840][2841][2842][2843][2844][2845][2846][2847][2848][2849][2850][2851][2852][2853][2854][2855][2856][2857][2858][2859][2860][2861][2862][2863][2864][2865][2866][2867][2868][2869][2870][2871][2872][2873][2874][2875][2876][2877][2878][2879][2880][2881][2882][2883][2884][2885][2886][2887][2888][2889][2890][2891][2892][2893][2894][2895][2896][2897][2898][2899][2900][2901][2902][2903][2904][2905][2906][2907][2908][2909][2910][2911][2912][2913][2914][2915][2916][2917][2918][2919][2920][2921][2922][2923][2924][2925][2926][2927][2928][2929][2930][2931][2932][2933][2934][2935][2936][2937][2938][2939][2940][2941][2942][2943][2944][2945][2946][2947][2948][2949][2950][2951][2952][2953][2954][2955][2956][2957][2958][2959][2960][2961][2962][2963][2964][2965][2966][2967][2968][2969][2970][2971][2972][2973][2974][2975][2976][2977][2978][2979][2980][2981][2982][2983][2984][2985][2986][2987][2988][2989][2990][2991][2992][2993][2994][2995][2996][2997][2998][2999][3000][3001][3002][3003][3004][3005][3006][3007][3008][3009][3010][3011][3012][3013][3014][3015][3016][3017][3018][3019][3020][3021][3022][3023][3024][3025][3026][3027][3028][3029][3030][3031][3032][3033][3034][3035][3036][3037][3038][3039][3040][3041][3042][3043][3044][3045][3046][3047][3048][3049][3050][3051][3052][3053][3054][3055][3056][3057][3058][3059][3060][3061][3062][3063][3064][3065][3066][3067][3068][3069][3070][3071][3072][3073][3074][3075][3076][3077][3078][3079][3080][3081][3082][3083][3084][3085][3086][3087][3088][3089][3090][3091][3092][3093][3094][3095][3096][3097][3098][3099][3100][3101][3102][3103][3104][3105][3106][3107][3108][3109][3110][3111][3112][3113][3114][3115][3116][3117][3118][3119][3120][3121][3122][3123][3124][3125][3126][3127][3128][3129][3130][3131][3132][3133][3134][3135][3136][3137][3138][3139][3140][3141][3142][3143][3144][3145][3146][3147][3148][3149][3150][3151][3152][3153][3154][3155][3156][3157][3158][3159][3160][3161][3162][3163][3164][3165][3166][3167][3168][3169][3170][3171][3172][3173][3174][3175][3176][3177][3178][3179][3180][3181][3182][3183][3184][3185][3186][3187][3188][3189][3190][3191][3192][3193][3194][3195][3196][3197][3198][3199][3200][3201][3202][3203][3204][3205][3206][3207][3208][3209][3210][3211][3212][3213][3214][3215][3216][3217][3218][3219][3220][3221][3222][3223][3224][3225][3226][3227][3228][3229][3230][3231][3232][3233][3234][3235][3236][3237][3238][3239][3240][3241][3242][3243][3244][3245][3246][3247][3248][3249][3250][3251][3252][3253][3254][3255][3256][3257][3258][3259][3260][3261][3262][3263][3264][3265][3266][3267][3268][3269][3270][3271][3272][3273][3274][3275][3276][3277][3278][3279][3280][3281][3282][3283][3284][3285][3286][3287][3288][3289][3290][3291][3292][3293][3294][3295][3296][3297][3298][3299][3300][3301][3302][3303][3304][3305][3306][3307][3308][3309][3310][3311][3312][3313][3314][3315][3316][3317][3318][3319][3320][3321][3322][3323][3324][3325][3326][3327][3328][3329][3330][3331][3332][3333][3334][3335][3336][3337][3338][3339][3340][3341][3342][3343][3344][3345][3346][3347][3348][3349][3350][3351][3352][3353][3354][3355][3356][3357][3358][3359][3360][3361][3362][3363][3364][3365][3366][3367][3368][3369][3370][3371][3372][3373
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [72 308.117 686.117 319.714]
xdvipdfmx:warning: Maybe incorrect paper size specified.
][3374][3375][3376][3377][3378][3379][3380][3381][3382][3383][3384][3385][3386][3387][3388][3389][3390][3391][3392][3393][3394][3395][3396][3397][3398][3399][3400][3401][3402][3403][3404][3405][3406][3407][3408][3409][3410][3411][3412][3413][3414][3415][3416][3417][3418][3419][3420][3421][3422][3423][3424][3425][3426][3427][3428][3429][3430][3431][3432][3433][3434][3435][3436][3437][3438][3439][3440
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [133.162 462.607 792.18 474.204]
xdvipdfmx:warning: Maybe incorrect paper size specified.
][3441][3442][3443][3444][3445][3446][3447][3448][3449][3450][3451][3452][3453][3454][3455][3456][3457][3458][3459][3460][3461][3462][3463][3464][3465][3466][3467][3468
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [133.162 647.912 733.364 659.509]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [133.162 522.383 624.892 533.98]
xdvipdfmx:warning: Maybe incorrect paper size specified.
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [133.162 486.517 720.95 498.114]
xdvipdfmx:warning: Maybe incorrect paper size specified.
][3469][3470][3471
xdvipdfmx:warning: Annotation out of page boundary.
xdvipdfmx:warning: Current page's MediaBox: [0 0 612 792]
xdvipdfmx:warning: Annotation: [133.162 247.414 779.284 259.011]
xdvipdfmx:warning: Maybe incorrect paper size specified.
][3472][3473][3474][3475][3476][3477][3478][3479][3480][3481][3482][3483][3484][3485][3486][3487][3488][3489][3490][3491][3492][3493][3494][3495][3496][3497][3498][3499][3500][3501][3502][3503][3504][3505][3506][3507][3508][3509][3510][3511][3512][3513][3514][3515][3516][3517][3518][3519][3520][3521][3522][3523][3524][3525][3526][3527][3528][3529][3530][3531][3532][3533][3534][3535][3536][3537][3538][3539][3540][3541][3542][3543][3544][3545][3546][3547][3548][3549][3550][3551][3552][3553][3554][3555][3556][3557][3558][3559][3560][3561][3562][3563][3564][3565][3566][3567][3568][3569][3570][3571][3572][3573][3574][3575][3576][3577][3578][3579][3580][3581][3582][3583][3584][3585][3586][3587][3588][3589][3590][3591][3592][3593][3594][3595][3596][3597][3598][3599][3600][3601][3602][3603][3604][3605][3606][3607][3608][3609]
xdvipdfmx:warning: Creating ToUnicode CMap failed for "/usr/local/texlive/2024/texmf-dist/fonts/opentype/public/fontawesome5/FontAwesome5Free-Solid-900.otf"
xdvipdfmx:warning: Failed to load ToUnicode CMap for font "/usr/local/texlive/2024/texmf-dist/fonts/opentype/public/fontawesome5/FontAwesome5Free-Solid-900.otf"

22866751 bytes written
Latexmk: Summary of warnings from last run of *latex:
  Latex failed to resolve 4 reference(s)
  =====Latex reported missing or unavailable character(s).
=====See log file for details.
Latexmk: ====Undefined refs and citations with line #s in .tex file:
  Missing character: There is no ^^C (U+0003) in font DejaVu Serif Bold/OT:script
  Missing character: There is no ^^C (U+0003) in font DejaVu Serif Bold/OT:script=latn;language=dflt;!
  Missing character: There is no ^^C (U+0003) in font DejaVu Serif Bold/OT:script
  Missing character: There is no ^^C (U+0003) in font DejaVu Serif Bold/OT:script=latn;language=dflt;!
  Missing character: There is no ^^C (U+0003) in font DejaVu Serif Bold/OT:script
  Missing character: There is no ^^C (U+0003) in font DejaVu Serif Bold/OT:script=latn;language=dflt;!
  Missing character: There is no ^^C (U+0003) in font DejaVu Serif Bold/OT:script
 And 57 more --- see log file 'sympy-1.14.dev.log'
Latexmk: All targets (sympy-1.14.dev.xdv sympy-1.14.dev.pdf) are up-to-date

make[1] : on quitte le répertoire « /pathtowork/sympy/doc/_build/latex »
```

Despite the various Latexmk warnings the return status is `0` of course.

asmeurer commented:
I can confirm the pdf builds on this branch. Let me know if you plan to push any more changes.

asmeurer commented:
OK I'm going to merge this. We can continue the other work in the other PRs.
